### PR TITLE
Length of Linked List in C, Java Added

### LINKED ISSUE #213: Length of a Linked List
Include the following two functions for determining the length of a linked list in these codes (https://github.com/jainaman224/Algo_Ds_Notes/tree/master/Linked_List)
- lengthIterative - This function should be implemented iteratively.
- lengthRecursive - This function should be implemented recursively.

Note : please specify the language in which you are interested before proceeding.

### PR DESCRIPTION
**Pull request by ravireddy07**
Fixes #213 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Length of linked Lists in C
-  Length of linked Lists in Java

#### Languages Used:
-  C
-  Java

#### Files Added:
- LengthOfLinkedList.c
- LengthOfLinkedList.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Linked_List/LengthOfLinkedList.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
+    #include<stdio.h>
```
Comment (jainaman224) [inline Linked_List/LengthOfLinkedList.c]: Incorrect indentation.

File: Linked_List/LengthOfLinkedList.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    */
+
+    struct Node {
+        	char data;
+	        struct Node* next;
+    };
+
+
+    void pushing_data_to_LL(struct Node** headPointer, char newData) {
+	        struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
```
Comment (jainaman224) [inline Linked_List/LengthOfLinkedList.c]: Use 4 space indentation.

File: Linked_List/LengthOfLinkedList.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    TestCase-1:
+    Sample Input: G S S O C 2 0 2 0
+    Sample Output: 9 //9 letters
+
+    TestCase-2:
+    Sample Input: O p e n - S o u r c e
+    Sample Output: 11 //11 letters
+
+    */
```
Comment (jainaman224) [inline Linked_List/LengthOfLinkedList.java]: Add a single new line at the end of file.

**Comment by ravireddy07** [comment]
@jainaman224  Sir, corrected the required changes. please check once.

File: Linked_List/LengthOfLinkedList.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    Sample Input: G S S O C 2 0 2 0 
+    Sample Output: 9 //9 letters
+
+    TestCase-2:
+    Sample Input: O p e n - S o u r c e
+    Sample Output: 11 //11 letters
+
+*/
+    
+
```
Comment (SakshiShreya) [inline Linked_List/LengthOfLinkedList.c]: Only 1 new-line should be there at the end of file.

Comment (ravireddy07) [reply Linked_List/LengthOfLinkedList.c]: removed that extra line at end and changes the file name according to the repo standards. Please check @SakshiShreya

File: Linked_List/Length_of_Linked_List.c [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    struct Node* current = head;
+    while(current != NULL)
+    {
+    	count++;
+    	current = current->next;
+    }
+    return count;
+}
+
+
```
Comment (somya-kapoor) [inline Linked_List/Length_of_Linked_List.c]: Please remove the extra line

File: Linked_List/Length_of_Linked_List.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+*/
+
+class Node
+{
+    String data;
+    Node next;
+    Node(String info)
+    {
+        data = info;
+	next = null;
```
Comment (somya-kapoor) [inline Linked_List/Length_of_Linked_List.java]: Please correct the indentation

File: Linked_List/Length_of_Linked_List.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        
+        thisList.pushing_data_to_LL("2");
+        thisList.pushing_data_to_LL("0");
+        thisList.pushing_data_to_LL("2");
+        thisList.pushing_data_to_LL("0");
+        
+        System.out.println("Currently the Length of the LinkedList is: " + thisList.gettingLength());
+    }
+}
+
```
Comment (somya-kapoor) [inline Linked_List/Length_of_Linked_List.java]: Please remove the extra line.

File: Linked_List/Length_of_Linked_List.c [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    This code is written for Char to store in LinkedList, if you want to store numericals or any other datatype,
+    just replace relevent data type in place of char in class Node.
+*/
+
+struct Node 
+{	
+    char data;
+    struct Node* next;
+};
+
```
Comment (somya-kapoor) [inline Linked_List/Length_of_Linked_List.c]: Please remove the extra line.
1 line is sufficient.

**Comment by ravireddy07** [comment]
@somya-kapoor  requested changes made. please check now.

File: Linked_List/Length_of_Linked_List.c [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+/*
+TestCase-1:
+Sample Input: G S S O C 2 0 2 0 
+Sample Output: 9 //9 letters
+
+TestCase-2:
+Sample Input: O p e n - S o u r c e
+Sample Output: 11 //11 letters
+*/
+
```
Comment (somya-kapoor) [inline Linked_List/Length_of_Linked_List.c]: Remove the line

**Comment by somya-kapoor** [comment]
@ravireddy07 Good work. After making those changes please squash the commits.

**Comment by ravireddy07** [comment]
done...changes made and squashed the commits. please check now. @somya-kapoor  @SakshiShreya  @jainaman224

**Pull request merged**